sw/airborne/modules/relative_localization_filter/discrete_ekf_no_north.c

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (C) Mario Coppola
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/relativelocalizationfilter/discrete_ekf_no_north.c"
+ * @author Steven van der Helm, Mario Coppola
+ * Discrete Extended Kalman Filter for Relative Localization without requiring knowledge of North
+ * This implementation is from
+ * van der Helm et al. "On-board Range-based Relative Localization for Micro Aerial Vehicles in indoor Leader-Follower Flight." (2018).
+ * Available at https://arxiv.org/pdf/1805.07171.pdf
+ */
+
+#include "discrete_ekf_no_north.h"
+#include "math/pprz_algebra_float.h"
+#include <math.h>
+#include <stdio.h> // needed for the printf statements
+
+enum ekf_statein {x12, y12, z1, z2, u1, v1, u2, v2, gam};
+enum ekf_input {u1dm, v1dm, u2dm, v2dm, r1m, r2m};
+
+void extractPhiGamma(float **inmat, float **phi, float **gamma, int m, int n_a, int n_b)
+{
+  int totalsize = m + n_b;
+  for (int i = 0; i < totalsize; i++) {
+    for (int j = 0; j < totalsize; j++) {
+      if (i < m && j < n_a) {
+        phi[i][j] = inmat[i][j];
+      } else if (i < m && j >= n_a) {
```

Comment:
This is equivalent to just "else if(i < m)". But more to the point, if you only want points with i < m why not just do for (int i = 0; i < m; i++) and not do these checks...